b_lambda_layer_common/source/python/b_lambda_layer_common/cache/data_cache.py

```diff
@@ -0,0 +1,80 @@
+import logging
+from typing import Any, Callable, Set, Union
+
+logger = logging.getLogger(__name__)
+
+try:
+    from b_lambda_layer_common.exceptions.container.internal_error import InternalError
+    from b_lambda_layer_common.ssm.refreshable import Refreshable
+except ImportError:
+    logger.exception(f'Failed import.')
+    from b_lambda_layer_common.source.python.b_lambda_layer_common.exceptions.container.internal_error import InternalError
+    from b_lambda_layer_common.source.python.b_lambda_layer_common.ssm.refreshable import Refreshable
+
+
+class DataCache(Refreshable):
+    """
+    Data cache implementation.
+    """
+
+    def __init__(self, max_age: int = 0):
+        """
+        Constructor.
+
+        :param max_age: Max age of the cache until reset is needed.
+        """
+        super().__init__(max_age)
+
+        self.__cache = {}
+
+    def update_value(self) -> None:
+        """
+        Force reset of the cache.
+
+        :return: No return.
+        """
+        self.__cache = {}
+
+    def use_cache(self, pointer: Union[str, Set[str]], func: Callable, *args, **kwargs) -> Any:
+        """
+        Cache function return values.
+
+        :param pointer: A string or a set of strings used to reference cached data.
+        :param func: A function whose return value will be cached.
+        :param args: Arguments that will be passed to func.
+
+        :key kwargs: Keyword arguments that will be passed to func.
+
+        :return: Cached data.
+        """
+        if self.should_refresh():
+            logger.info('Cached data should be reset.')
```

Comment:
When logging what is going to happen, use continuous present tense. Then if an error follows, you will know what action was being performed.
For example: Resetting cached data...

Reply:
Don't forget the three dots ...